src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SubqueryIT.java

```diff
@@ -267,4 +274,101 @@ public void nonCorrelatedNotExistsParentWhere() throws IOException {
                 )
         );
     }
+
+    @Test
+    public void selectFromSubqueryWithCountShouldPass() throws IOException {
+        JSONObject result = executeQuery(
+                String.format("SELECT t.TEMP as count " +
+                              "FROM (SELECT COUNT(*) as TEMP FROM %s/account) t", TEST_INDEX_ACCOUNT));
+
+        assertThat(result.query("/aggregations/count/value"), equalTo(1000));
+    }
+
+    @Test
+    public void selectFromSubqueryWithWhereAndCountShouldPass() throws IOException {
+        JSONObject result = executeQuery(
+                String.format("SELECT t.TEMP as count " +
+                              "FROM (SELECT COUNT(*) as TEMP FROM %s/account WHERE age > 30) t", TEST_INDEX_ACCOUNT));
+
+        assertThat(result.query("/aggregations/count/value"), equalTo(502));
+    }
+
+    @Test
+    public void selectFromSubqueryWithCountAndGroupByShouldPass() throws Exception {
+        JSONObject result = executeQuery(
+                String.format("SELECT t.TEMP as count " +
+                              "FROM (SELECT COUNT(*) as TEMP FROM %s/account GROUP BY gender) t", TEST_INDEX_ACCOUNT));
+
+        assertThat(getTotalHits(result), equalTo(1000));
+        JSONObject gender = (JSONObject) result.query("/aggregations/gender");
+        assertThat(gender.getJSONArray("buckets").length(), equalTo(2));
+
+        boolean isMaleFirst = gender.optQuery("/buckets/0/key").equals("m");
+        int maleBucketId = isMaleFirst ? 0 : 1;
+        int femaleBucketId = isMaleFirst ? 1 : 0;
+
+        String maleBucketPrefix = String.format(Locale.ROOT, "/buckets/%d", maleBucketId);
+        String femaleBucketPrefix = String.format(Locale.ROOT, "/buckets/%d", femaleBucketId);
+
+        assertThat(gender.query(maleBucketPrefix + "/key"), equalTo("m"));
+        assertThat(gender.query(maleBucketPrefix + "/count/value"), equalTo(507));
+        assertThat(gender.query(femaleBucketPrefix + "/key"), equalTo("f"));
+        assertThat(gender.query(femaleBucketPrefix + "/count/value"), equalTo(493));
+    }
+
+    @Test
+    public void selectFromSubqueryWithCountAndGroupByAndOrderByShouldPass() throws IOException {
+        JSONObject result = executeQuery(
+                String.format(
```

Comment:
Should we use StringUtils instead of String here?

Reply:
Done.